sdk/core/core-http/lib/policies/tracingPolicy.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { TracerProxy, TraceOptions } from "@azure/core-tracing";
+import {
+  RequestPolicyFactory,
+  RequestPolicy,
+  RequestPolicyOptions,
+  BaseRequestPolicy
+} from "./requestPolicy";
+import { WebResource } from "../webResource";
+import { HttpOperationResponse } from "../httpOperationResponse";
+
+export function tracingPolicy(): RequestPolicyFactory {
+  return {
+    create(nextPolicy: RequestPolicy, options: RequestPolicyOptions) {
+      return new TracingPolicy(nextPolicy, options);
+    }
+  };
+}
+
+export class TracingPolicy extends BaseRequestPolicy {
+  constructor(nextPolicy: RequestPolicy, options: RequestPolicyOptions) {
+    super(nextPolicy, options);
+  }
+
+  public async sendRequest(request: WebResource): Promise<HttpOperationResponse> {
+    if (!request.spanOptions || !request.spanOptions.parent) {
+      return this._nextPolicy.sendRequest(request);
+    }
+
+    // create a new span
+    const tracer = TracerProxy.getTracer();
+    const span = tracer.startSpan("core-http", request.spanOptions);
+
+    span.start();
+
+    try {
+      // set headers
+      const spanContext = span.context();
```

Comment:
I think I'll need to update the NoOpSpan to stop throwing errors when methods are called and instead be treated as no-op. As it is now if `getTracer()` returned a NoOpTracerPlugin and a span was passed in `spanOptions`, this would throw since the no-op `span.context()` currently throws. I'll add a test for this scenario as well.

Reply:
Yep, NoOp should definitely not throw when called

Reply:
Updated!